src/Models/BaseElement.php

```diff
@@ -712,35 +718,65 @@ public function CMSEditLink($directLink = false)
             return $this->cacheData['cms_edit_link'];
         }
 
+        $link = $this->getElementCMSLink($directLink);
+        $this->extend('updateCMSEditLink', $link);
+
+        if ($link) {
+            $this->cacheData['cms_edit_link'] = $link;
+        }
+
+        return $link;
+    }
+
+    /**
+     * @param bool $directLink
+     * @return null|string
+     */
+    private function getElementCMSLink(bool $directLink)
+    {
         $relationName = $this->getAreaRelationName();
         $page = $this->getPage();
 
+        $link = null;
+
         if (!$page) {
-            $link = null;
-            $this->extend('updateCMSEditLink', $link);
             return $link;
         }
-
-        if (!$page instanceof SiteTree && method_exists($page, 'CMSEditLink')) {
-            $link = Controller::join_links($page->CMSEditLink(), 'ItemEditForm');
-        } else {
+        // If $page is a Page
+        if ($page instanceof SiteTree) {
             $link = $page->CMSEditLink();
+        } else {
+            // Envoke method only if it exists in the $page
+            if (ClassInfo::hasMethod($page, 'CMSEditLink')) {
```

Comment:
```suggestion
        } else if (ClassInfo::hasMethod($page, 'CMSEditLink')) {
```

Reply:
Just in case, I use ELSEIF instead ELSE IF, because of this warning "Usage of ELSE IF is discouraged; use ELSEIF instead"

Reply:
Yup, fair enough. 👍